source/common/http/filter_manager.h

```diff
@@ -715,6 +718,12 @@ class FilterManager : public ScopeTrackedObject,
     for (const auto& log_handler : access_log_handlers_) {
       log_handler->log(log_context, streamInfo());
     }
+    if (Runtime::runtimeFeatureEnabled(
+            "envoy.reloadable_features.http_separate_config_and_filter_access_loggers")) {
+      for (const auto& log_handler : config_log_handlers_) {
+        log_handler->log(log_context, streamInfo());
+      }
+    }
```

Comment:
Things actually could be much simpler. We actually needn't to store the access log handlers from configuration to a list because they are managed by the filter chain configuration, right?

So, you actually could call evaluate the access log handlers from configuration at the `ActiveStream`. Just for example, you can create a new method `log()` for `ActiveStream`:

```
    void log(AccessLog::AccessLogType type) {
      const Formatter::HttpFormatterContext log_context{
          request_headers_.get(), response_headers_.get(), response_trailers_.get(), {}, type,
          active_span_.get()};

      const bool filter_access_loggers_first =
          Runtime::runtimeFeatureEnabled("envoy.reloadable_features.filter_access_loggers_first");

      if (!filter_access_loggers_first) {
        for (const auto& access_logger : connection_manager_.config_->accessLogs()) {
          access_logger->log(log_context, filter_manager_.streamInfo());
        }
      }

      filter_manager_.log(log_context);

      if (filter_access_loggers_first) {
        for (const auto& access_logger : connection_manager_.config_->accessLogs()) {
          access_logger->log(log_context, filter_manager_.streamInfo());
        }
      }
    }

```

Reply:
/wait